kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DatasetRepository.java

```diff
@@ -181,6 +181,52 @@
   <E> Dataset<E> update(String namespace, String name, DatasetDescriptor descriptor);
 
   /**
+   * Update a dataset to indicate that a set of constraints
```

Comment:
@ben-roling and I debated if adding the API at this level was correct, we explored some other options like adding a protected scope method to the AbstractDatasetRepository to allow implementations of Datasets to gain access to the MetadataProvider to call these methods there rather than adding a mirror of those methods here on the DatasetRepository.

We went with the mirrored methods for the first pass since this seemed to be a more established pattern with other methods nearly matching between the DatasetRepository and MetadataProvider (the create and update methods being examples, although their return values are different, Dataset vs DatasetDescriptors)